src/components/search/Suggestions/Suggestions.tsx

```diff
@@ -0,0 +1,146 @@
+import type { HTMLAttributes } from 'react'
+import React, { forwardRef } from 'react'
+import { List as UIList } from '@faststore/ui'
+import Button from 'src/components/ui/Button'
+import Link from 'src/components/ui/Link'
+
+import SuggestionProductCard from '../SuggestionProductCard'
+
+import './suggestions.scss'
+
+const MAX_SUGGESTIONS = 10
+const MAX_SUGGESTIONS_WITH_PRODUCTS = 5
+const MAX_SUGGESTED_PRODUCTS = 4
+const SUGGESTED_PRODUCTS = [
+  {
+    name: 'Ergonomic Wooden Bacon',
+    listPrice: 72.06,
+    price: 46.26,
+    image: [
+      {
+        alternateName: 'rerum',
+        url: 'http://storeframework.vtexassets.com/arquivos/ids/167285/ut.jpg?v=637753017045600000',
+      },
+    ],
+  },
+  {
+    name: 'Handcrafted Rubber Sausages',
+    listPrice: 59.57,
+    price: 32.83,
+    image: [
+      {
+        alternateName: 'ea',
+        url: 'http://storeframework.vtexassets.com/arquivos/ids/155949/voluptas.jpg?v=637752878341070000',
+      },
+    ],
+  },
+]
+
+const SUGGESTIONS = ['Sony MX', 'Sony MV-100 Headphone', 'Sony M2000 Earbuds']
+
+function formatSearchTerm(
+  indexSubstring: number,
+  searchTerm: string,
+  suggestion: string
+) {
+  if (indexSubstring === 0) {
+    return searchTerm
+      .split('')
+      .map((char, idx) =>
+        idx === 0 && suggestion.indexOf(char.toUpperCase()) === 0
+          ? char.toUpperCase()
+          : char.toLowerCase()
+      )
+      .join('')
+  }
+
+  return searchTerm.toLowerCase()
+}
+
+export interface SuggestionsProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * Search term
+   */
+  term?: string
+}
+
+const Suggestions = forwardRef<HTMLDivElement, SuggestionsProps>(
+  function Suggestions(
+    { testId = 'suggestions', term = '', ...otherProps },
+    ref
+  ) {
+    const suggestions =
+      SUGGESTED_PRODUCTS.length > 0
+        ? SUGGESTIONS.slice(0, MAX_SUGGESTIONS_WITH_PRODUCTS)
+        : SUGGESTIONS.slice(0, MAX_SUGGESTIONS)
+
+    const handleSuggestions = (suggestion: string, searchTerm: string) => {
```

Comment:
Could this also be considered a functional component (because it is returning some elements) instead of just a function?

Do you know if there's any penalty in defining it inside another component or if it's best to define it outside?